.github/workflows/java-cd.yml

```diff
@@ -149,12 +127,41 @@ jobs:
                   jar -cvf bundle.jar *
                   ls -ltr
                   cd -
-                  cp $src_folder/bundle.jar bundle-${{ matrix.host.TARGET }}.jar
+                  cp $src_folder/bundle.jar bundle-${{ matrix.build.TARGET }}.jar
 
             - name: Upload artifacts to publish
               continue-on-error: true
               uses: actions/upload-artifact@v4
               with:
-                  name: java-${{ matrix.host.TARGET }}
+                  name: java-${{ matrix.build.TARGET }}
                   path: |
                       java/bundle*.jar
+
+    combine-binaries-to-one-directory:
+        needs: create-binaries-to-publish
+        runs-on: ubuntu-latest
+        env:
+            INPUT_VERSION: ${{ github.event.inputs.version }}
+        steps:
+            - name: Download published artifacts
+              uses: actions/download-artifact@v4
+
+            - name: Display structure of downloaded files
+              run: ls -R
+
+            - name: Move all files to one zip file
+            # TODO: Add `./java-aarch64-unknown-linux-gnu/* bundles` when we are able to support a linux arm runner.
+            # TODO: Add "maven" to linux arm PACKAGE_MANAGER in build-matrix.json.
+              run: |
+                mkdir bundles
+                mv ./java-aarch64-apple-darwin/* bundles
+                mv ./java-x86_64-apple-darwin/* bundles
+                mv ./java-x86_64-unknown-linux-gnu/* bundles
+                zip -r bundles-${{ env.INPUT_VERSION }}.zip bundles
```

Comment:
Don't zip - GHA also zips, so you have a zip inside zip
![image](https://github.com/user-attachments/assets/e48c330c-f28d-43dc-a996-2f743a50f459)